src/types.ts

```diff
@@ -0,0 +1,27 @@
+import type { Readable } from 'node:stream';
+import type { Dispatcher, FormData } from 'undici'
+import type { IncomingHttpHeaders } from 'undici/types/header.js';
+import type BodyReadable from 'undici/types/readable.js';
+
+export type RequestOptions = { // partially derived from undici.Dispatcher.RequestOptions
+    url: string;
+    method: Dispatcher.HttpMethod;
+    /** 
+     * @default null 
+     * */
+    body?: string | Buffer | Uint8Array | Readable | null | FormData;
+    /**
+     * Headers to be sent with the request.
+     * 
+     * @default new Headers()
+     */
+    headers?: Headers;
+}
+
+export type ResponseData = { // partially derived from undici.Dispatcher.ResponseData
+    statusCode: number;
+    headers: IncomingHttpHeaders;
```

Comment:
```suggestion
    headers: Dispatcher.ResponseData['headers']
```

I'd be more comfortable relying on a nested type than importing a file that isn't explicitly listed in exports